src/arb/ArbTypes.jl

```diff
@@ -1377,3 +1377,24 @@ function _acb_mat_clear_fn(x::AcbMatrix)
   ccall((:acb_mat_clear, libflint), Nothing, (Ref{AcbMatrix}, ), x)
 end
 
+
+################################################################################
+#
+#   Type unions
+#
+################################################################################
+
+const RealFieldElemOrPtr = Union{RealFieldElem, Ref{RealFieldElem}, Ptr{RealFieldElem}}
+const ArbFieldElemOrPtr = Union{ArbFieldElem, Ref{ArbFieldElem}, Ptr{ArbFieldElem}}
```

Comment:
have you considered adding a function `f(T) = Union{T, Ref{T}, Ptr{T}}` or even a macro `@add_ptr_ref_unino ZZRingElem` which expands to this (or equivalently an `@eval` for loop).

The drawback of using macros is that one can't simply grep for `ZZRingElemOrPtr` to find its definition...

Anyway, I am also OK with this as-is.